vpr/test/test_edge_groups.cpp

```diff
@@ -0,0 +1,31 @@
+#include <array>
+#include <utility>
+#include <cstddef>
+#include <set>
+
+#include "catch.hpp"
+
+#include "rr_graph.h"
+
+namespace {
+
+TEST_CASE("edge_groups_create_sets", "[vpr]") {
```

Comment:
Does this test case fail under the previous logic?

Reply:
It should, but actually testing it requires backporting my test. I'll do that.

Reply:
Thanks, tt didn't fail. I made a better test that does. Here's the backported test: https://github.com/HackerFoo/vtr-verilog-to-routing/tree/test-edge-groups-test

Reply:
Great, thanks for checking that.